spec/acceptance/self_service_spec.rb

```diff
@@ -262,6 +263,25 @@ class {'self_service':
         expect(result.stdout).to match(%r{false})
         run_shell('touch -d "1 day ago"  /opt/puppetlabs/server/pe_build')
       end
+      context 'S0035' do
+        # verify pre-conditions are true
+        it 'returns true before adding an invalid module' do
+          result = run_shell('facter -p self_service.S0035')
+          expect(result.stdout).to match(%r{true})
+        end
+
+        # act installing an invalid module
+        run_shell('puppet module install puppetlabs-ntp --version 9.1.0; puppet module install puppetlabs-stdlib --version 0.1.5 --force')
```

Comment:
This is outside of an `it` block. It will not run in the order you expect unless it is within the `it` block. It needs to be in a `before` or in the `it` block. Like the following. 

```
   context 'S0035' do
        # verify pre-conditions are true
        it 'returns true before adding an invalid module' do
          result = run_shell('facter -p self_service.S0035')
          expect(result.stdout).to match(%r{true})
        end

        # verify false after action
        it 'returns false after adding an invalid module' do
          # act installing an invalid module
          run_shell('puppet module install puppetlabs-ntp --version 9.1.0; puppet module install puppetlabs-stdlib --version 0.1.5 --force')
          result = run_shell('facter -p self_service.S0035')
          expect(result.stdout).to match(%r{false})
          run_shell('puppet module uninstall puppetlabs-ntp--force')
          run_shell('puppet module uninstall puppetlabs-stdlib --force')
        end
      end
```